### ISSUE TITLE
Uniform/Int distribution does not check formula

### ISSUE THREAD

**Issue by assignUser** [kgoldfeld/simstudy#15]
There is no error, just produces NAs.
```r
 def <- defData( varname = "x1", dist = "uniformInt", formula = "30;20")
genData(5,def)
 
   id x1
1:  1 NA
2:  2 NA
3:  3 NA
```

**Comment by kgoldfeld** [comment]
Thanks - that needs to be fixed - I'll make sure to flag this an error, since min cannot exceed max.  I'll wait until you find more errors.

I'd love to hear how you are developing these tests ...

**Comment by assignUser** [comment]
Sorry for the late reply, I'm pretty into it right now but gave into my usual weakness. Namely starting big not small, so I have nothing finished yet xD I will show you soon though.

You can mark this issue with the "bug"/"error" label. If I find any additional issues I will create new ones, with the intent of keeping a record and fixing them later, not pawning them of to you!

**Comment by kgoldfeld** [comment]
Sounds good. I don't mind fixing the bugs, but am very happy to have you work on it.


Are you getting credit for this, or just interested in getting involved? As you can probably tell, I am not a professional programmer, and there are decisions I made early on that I wish I could change, but unfortunately cannot given that there are thousands of users out there. In any case, I am very happy to have someone to help me fix, improvi, and grow this package.

________________________________

**Comment by assignUser** [comment]
I just noticed that I ruled out variables in formulas for the uniform/Int and categorical distributions with the way I wrote the new evalDef. Previously as far as I can see this should be possible, so I will change it to keep the same functionality!

As an update: I am currently working on test coverage for the generator functions and modifying them to work with external variables.

**Comment by assignUser** [comment]
For consistency I have added the same behavior to .gencat for probabilities != 1 that catProbs uses.

**Comment by assignUser** [comment]
Closed by #22
